pkg/datapath/loader/loader.go

```diff
@@ -185,7 +247,17 @@ func patchHostNetdevDatapath(ep datapath.Endpoint, objPath, dstPath, ifName stri
 	callsMapHostDevice := bpf.LocalMapName(callsmap.HostMapName, uint16(ep.GetID()))
 	strings[callsMapHostDevice] = bpf.LocalMapName(callsmap.NetdevMapName, uint16(ifIndex))
 
-	return hostObj.Write(dstPath, opts, strings)
+	// Write object and header files specific to ifName.
+	netdevHeaderFile, err := patchHostNetdevHeaderFile(ep.StateDir(), ifName, mac,
+		ifIndex, opts["SECCTX_FROM_IPCACHE"], option.Config.EnableNodePort)
+	if err != nil {
+		return err
+	}
+	err = hostObj.Write(dstPath, opts, strings)
+	if err != nil {
+		os.Remove(netdevHeaderFile)
```

Comment:
Is there a reason that we don't write the header file after the object file? In this case, we wouldn't need to remove the header file in case of an error

Reply:
Related question: Should we error out when writing the header fails? (If we should, then I guess swapping header and object order doesn't change much, since you might want to clean up the object file if writing the header failed?)

From the commit log I understand that a missing header would not technically be fatal, although it makes sense to me to error out in case those distinct headers became necessary for some reason in the future?

Reply:
You are both correct. That is, a missing header shouldn't be fatal and a warning is likely enough, thus we can change the order.